firebase-sessions/src/test/kotlin/com/google/firebase/sessions/SessionCoordinatorTest.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.google.firebase.sessions.testing.FakeFirebaseInstallations
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+class SessionCoordinatorTest {
+  @Test
+  fun attemptLoggingSessionEvent_populatesFid() = runTest {
+    val sessionCoordinator =
+      SessionCoordinator(
+        firebaseInstallations = FakeFirebaseInstallations("FaKeFiD"),
+        backgroundDispatcher = StandardTestDispatcher(testScheduler),
```

Comment:
`:integ-testing` doesn't have any test `CoroutineDispatcher`s to work well with `runTesting`.

Reply:
`:integ-testing` is to provide executors for testing that mimic the executors provided by `firebase-common` with policies e.g. don't do blocking network requests on Background. See https://firebase.github.io/firebase-android-sdk/components/executors/#testing

So these tests don't mimic the Background policy. I tried to use `TestOnlyExecutors.background().asCoroutineDispatcher()` when I first wrote this test, but then I couldn't call `runCurrent()` to wait for the suspend functions to finish before doing the assert.